server/src/test/java/org/elasticsearch/snapshots/SnapshotsServiceTests.java

```diff
@@ -0,0 +1,526 @@
+            final Environment environment = new Environment(settings, createTempDir());
+            nodeEnv = new NodeEnvironment(settings, environment);
+            final NamedXContentRegistry namedXContentRegistry = new NamedXContentRegistry(Collections.emptyList());
+            final ScriptService scriptService = new ScriptService(settings, emptyMap(), emptyMap());
+            client = new NodeClient(settings, threadPool);
+            final IndexScopedSettings indexScopedSettings =
+                new IndexScopedSettings(settings, IndexScopedSettings.BUILT_IN_INDEX_SETTINGS);
+            indicesService = new IndicesService(
+                settings,
+                mock(PluginsService.class),
+                nodeEnv,
+                namedXContentRegistry,
+                new AnalysisRegistry(environment, emptyMap(), emptyMap(), emptyMap(), emptyMap(), emptyMap(),
+                    emptyMap(), emptyMap(), emptyMap(), emptyMap()),
+                indexNameExpressionResolver,
+                new MapperRegistry(emptyMap(), emptyMap(), MapperPlugin.NOOP_FIELD_FILTER),
+                new NamedWriteableRegistry(Collections.emptyList()),
+                threadPool,
+                indexScopedSettings,
+                new NoneCircuitBreakerService(),
+                new BigArrays(new PageCacheRecycler(settings), null, "test"),
+                scriptService,
+                client,
+                new MetaStateService(nodeEnv, namedXContentRegistry),
+                Collections.emptyList(),
+                emptyMap()
+            );
+            final RecoverySettings recoverySettings = new RecoverySettings(settings, clusterSettings);
+            final ActionFilters actionFilters = new ActionFilters(emptySet());
+            snapshotShardsService = new SnapshotShardsService(
+                settings, clusterService, snapshotsService, threadPool,
+                transportService, indicesService, actionFilters, indexNameExpressionResolver);
+            final ShardStateAction shardStateAction = new ShardStateAction(
+                clusterService, transportService, allocationService,
+                new RoutingService(settings, clusterService, allocationService),
+                deterministicTaskQueue.getThreadPool()
+            );
+            indicesClusterStateService = new IndicesClusterStateService(
+                settings, indicesService, clusterService, threadPool,
+                new PeerRecoveryTargetService(
+                    deterministicTaskQueue.getThreadPool(), transportService, recoverySettings,
+                    clusterService
+                ),
+                shardStateAction,
+                new NodeMappingRefreshAction(transportService, new MetaDataMappingService(clusterService, indicesService)),
+                repositoriesService,
+                mock(SearchService.class),
```

Comment:
Just mocked this one out since it's not relevant for the test and just a bunch of code to get up and running.